src/main/java/bio/terra/service/tabulardata/google/BigQueryPdao.java

```diff
@@ -435,6 +436,105 @@ public void createSnapshot(Snapshot snapshot, List<String> rowIds) throws Interr
         snapshotViewCreation(datasetBqDatasetName, snapshotName, snapshot, projectId, bigQuery, bigQueryProject);
     }
 
+    private static final String insertAllLiveViewDataTemplate =
+        "INSERT INTO `<project>.<snapshot>.<dataRepoTable>` " +
+            "(<dataRepoTableId>, <dataRepoRowId>) <liveViewTables>";
+
+    private static final String getLiveViewTableTemplate =
+        // TODO pull insert out and loop thru rest w UNION ()
+            "(SELECT '<tableId>', <dataRepoRowId> FROM `<project>.<dataset>.<liveView>`)";
+
+    private static final String mergeLiveViewTablesTemplate =
+        "<selectStatement1> UNION ALL <selectStatement2>";
+
+    private static final String validateSnapshotSizeTemplate =
+        "SELECT * FROM <project>.<snapshot>.<dataRepoTable>";
+
+    public String createSnaphotTableFromLiveViews(
+        BigQueryProject bigQueryProject,
+        List<DatasetTable> tables,
+        String datasetBqDatasetName) throws InterruptedException {
+
+        String output = "";
+
+        // recursively call the merge template to spool together all the different select statements
+        if (tables.size() == 1) {
+            DatasetTable onlyTable = tables.get(0);
+            TableId liveViewId = TableId.of(datasetBqDatasetName, onlyTable.getName());
+            String liveViewTableName = liveViewId.getTable();
```

Comment:
Nitpick: `onlyTable.getName()` should already == the name of the live view, I don't think this is needed.